prime-router/src/main/kotlin/metadata/Mappers.kt

```diff
@@ -468,16 +469,20 @@ class LookupMapper : Mapper {
         sender: Sender?
     ): ElementResult {
         return ElementResult(
-            if (values.size != args.size) {
+            // args should be twice size of values as it includes the index column names
+            if (values.size != args.size.div(2)) {
                 null
             } else {
                 val lookupTable = element.tableRef
                     ?: error("Schema Error: could not find table ${element.table}")
                 val tableFilter = lookupTable.FilterBuilder()
-                values.forEach {
-                    val indexColumn = it.element.tableColumn
-                        ?: error("Schema Error: no tableColumn for element ${it.element.name}")
-                    tableFilter.equalsIgnoreCase(indexColumn, it.value)
+                values.forEachIndexed { index, elementAndValue ->
+                    // retrieve column name to use for lookup.
+                    // Specified in sender settings lookup mapper parameters
+                    val indexColumn = args[index.times(2).plus(1)].split(":")[1]
+                    if (indexColumn.isEmpty())
+                        "Schema Error: no tableColumn for element ${elementAndValue.element.name}"
```

Comment:
Think this should be `error("Schema Error: no tableColumn for element ${elementAndValue.element.name}"`?